Source/WebCore/svg/SVGPathElement.cpp

```diff
@@ -37,15 +37,59 @@ namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(SVGPathElement);
 
-static constexpr uint64_t maxPathSegListCacheSize = 150 * 1024; // 150 Kb.
-static uint64_t pathSegListCacheSize = 0;
+class PathSegListCache {
+public:
+    static PathSegListCache& singleton();
 
-static HashMap<AtomString, SVGPathByteStream::Data>& pathSegListCache()
+    const SVGPathByteStream::Data* get(const AtomString& attributeValue) const;
+    void add(const AtomString& attributeValue, const SVGPathByteStream::Data&);
+    void clear();
+
+private:
+    friend class NeverDestroyed<PathSegListCache, MainThreadAccessTraits>;
+    PathSegListCache() = default;
+
+    HashMap<AtomString, SVGPathByteStream::Data> m_cache;
+    uint64_t m_sizeInBytes { 0 };
+    static constexpr uint64_t maxSizeInBytes = 150 * 1024; // 150 Kb.
```

Comment:
This is the size of data only. The keys can be really long strings. I think the total size of the HashMap can be double this size. Is it expected to have a cache of 300kb for the path data cache?

Reply:
We can have this discussion but this patch is about urgently fixing a crash that needs to be cherry-picked to the branch ASAP. I did not change the cache size in this patch.

The short answer is that the size was chosen was Speedometer 3 content. We can change it but I think this is out of scope for this patch. I already refactored the code as you suggested even though I prefer to keep branch fixes as simple as possible.